client/src/pages/Home.js

```diff
@@ -110,17 +110,16 @@ export default function Home({
         <label>Filter group:</label>
         <select value={groupValue} onChange={filterGroups}>
           <option>Please select...</option>
-          {availableGroups.map((group) => (
-            <option>{group}</option>
+          {availableGroups.map((group, index) => (
+            <option key={index}>{group}</option>
```

Comment:
```suggestion
          {availableGroups.map((group) => (
            <option key={group._id}>{group}</option>
```
Now that you have `group` objects and hence access to the `_id` property I would suggest using that instead of the `index`.